e2e/tests/performance/notebook.perf.spec.js

```diff
@@ -0,0 +1,153 @@
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+/*
+This test suite is dedicated to performance tests to ensure that testability of performance
+is not broken upstream on Open MCT. Any assumptions made downstream will be tested here
+
+TODO:
+ - Update resolution of performance config
+ - Add Performance Observer on init to push all performance marks
+ - Move client CDP connection to before or to a fixture
+ -
+
+*/
+
+const { test, expect } = require('@playwright/test');
+
+const notebookFilePath = 'e2e/test-data/PerformanceNotebook.json';
+
+test.describe('Performance tests', () => {
+    test.beforeEach(async ({ page, browser }, testInfo) => {
+        // Go to baseURL
+        await page.goto('/', { waitUntil: 'networkidle' });
+
+        // Click a:has-text("My Items")
+        await page.locator('a:has-text("My Items")').click({
+            button: 'right'
+        });
+
+        // Click text=Import from JSON
+        await page.locator('text=Import from JSON').click();
+
+        // Upload Performance Display Layout.json
+        await page.setInputFiles('#fileElem', notebookFilePath);
+
+        // TODO Fix this
+        await page.locator('text=OK >> nth=1').click();
+
+        await expect(page.locator('a:has-text("Performance Notebook")')).toBeVisible();
+
+        //Create a Chrome Performance Timeline trace to store as a test artifact
+        console.log("\n==== Devtools: startTracing ====\n");
+        await browser.startTracing(page, {
+            path: `${testInfo.outputPath()}trace.json`,
+            screenshots: true
+        });
+    });
+    test.afterEach(async ({ page, browser}) => {
+        console.log("\n==== Devtools: stopTracing ====\n");
+        await browser.stopTracing();
+
+        /* Measurement Section
+        / The following section includes a block of performance measurements.
+        */
+        //Get All Performance Marks
+        const getAllMarksJson = await page.evaluate(() => JSON.stringify(window.performance.getEntriesByType("mark")));
+        const getAllMarks = JSON.parse(getAllMarksJson);
+        console.log('window.performance.getEntriesByType("mark")', getAllMarks);
+
+        //Get All Performance Measures
+        const getAllMeasuresJson = await page.evaluate(() => JSON.stringify(window.performance.getEntriesByType("measure")));
+        const getAllMeasures = JSON.parse(getAllMeasuresJson);
+        console.log('window.performance.getEntriesByType("measure")', getAllMeasures);
+
+    });
+    /* The following test will navigate to a previously created Performance Display Layout and measure the
+    /  following metrics:
+    /  - ElementResourceTiming
+    /  - Interaction Timing
+    */
+    test.only('Notebook Search, Add Entry, Update Entry are performant', async ({ page, browser }) => {
```

Comment:
I think search will bear fruit when testing on Couch (like we discussed), but for adding and updating entries, while these are saved on the backend, I _believe_ they are updated immediately on the front end, so there may not be much gained from those.

Reply:
@jvigliotta absolutely, this is mostly to measure how the user's performance timing is adjust as we change the application